repos/system_upgrade/cloudlinux/actors/handleelspackages/files/els_script_handler.sh

```diff
@@ -0,0 +1,130 @@
+RUNNING_KERNEL=$(uname -r)
+
+echo "Starting ELS package handling"
+echo "Running kernel: ${RUNNING_KERNEL}"
+
+mkdir -p "${BACKUP}" 2>/dev/null || {
+    echo "Failed to create backup directory ${BACKUP}"
+    exit 1
+}
+
+# Check if ELS repositories are present
+if ! ls /etc/yum.repos.d/centos*els*.repo &> /dev/null; then
+    echo "No ELS repositories found, skipping processing"
+    exit 0
+fi
+
+echo "ELS repositories detected. Processing ELS packages before conversion..."
+
+# Create package backup list
+rpm -qa | sort > "${BACKUP}/els_packages_backup_$(date +%Y%m%d_%H%M%S).txt" || {
+    echo "Warning: Failed to create package backup list"
+}
+
+# Function to check if kernel is in use
+is_kernel_in_use() {
+    local kernel_pkg=$1
+    local kernel_ver=$(rpm -q --queryformat '%{VERSION}-%{RELEASE}' "$kernel_pkg")
+    [[ "$kernel_ver" == "$RUNNING_KERNEL" ]]
+}
+
+# Prevent ELS packages from being reinstalled
+if ! grep -q "exclude=\*.tuxcare.els\*" /etc/yum.conf; then
+    {
+        echo
+        echo "# Added by cloudlinux Elevate"
+        echo "exclude=*.tuxcare.els*"
+    } >> /etc/yum.conf || echo "Warning: Failed to add exclusion to yum.conf"
+fi
+
+# Save and disable ELS repos
+cp /etc/yum.repos.d/centos*els*.repo "${BACKUP}/" 2>&1
+echo "Disabling ELS repositories..."
+yum-config-manager --disable centos7-els 2>&1
+for i in {1..6}; do
+    yum-config-manager --disable centos7els-rollout-$i 2>&1
+done
+
+# Remove els-define package first if present
+if rpm -q els-define &> /dev/null; then
+    yum remove -y els-define 2>&1 || echo "Warning: Failed to remove els-define"
+fi
+
+# Get list of installed ELS packages
+echo "Checking for ELS packages..."
+els_pkgs=$(rpm -qa | grep -E '\.tuxcare\.els[0-9]') || true
+
+if [ -n "$els_pkgs" ]; then
+    echo "Found ELS packages:"
+    echo "$els_pkgs"
+
+    # Handle kernel packages separately and more aggressively
+    els_kernel_pkgs=$(echo "$els_pkgs" | grep "^kernel") || true
+    if [ -n "$els_kernel_pkgs" ]; then
+        echo "Found ELS kernel packages:"
+        echo "$els_kernel_pkgs"
+
+        # Remove all ELS kernel packages, regardless of running kernel
+        # since we already have the LVE kernel installed
+        echo "Removing all ELS kernel packages since LVE kernel is present..."
+        for kernel_pkg in $els_kernel_pkgs; do
+            echo "Removing ELS kernel package: $kernel_pkg"
+            rpm -e --nodeps "$kernel_pkg" 2>&1 || {
+                echo "Warning: Failed to remove $kernel_pkg"
+                # If rpm -e fails, try with yum as fallback
+                yum remove -y "$kernel_pkg" 2>&1
+            }
+        done
+    fi
+
+    # Process remaining non-kernel ELS packages
+    non_kernel_els_pkgs=$(echo "$els_pkgs" | grep -v "^kernel") || true
+    if [ -n "$non_kernel_els_pkgs" ]; then
+        echo "Processing non-kernel ELS packages..."
+
+        # Create a yum shell script for batch processing
+        TMPFILE=$(mktemp)
```

Comment:
Don't use /tmp for secutiry reasons please, any directory which has only root:root write access should be ok.